bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveCRC16EncapsulationCommandClass.java

```diff
@@ -0,0 +1,172 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zwave.internal.protocol.commandclass;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+import org.openhab.binding.zwave.internal.protocol.SerialMessage;
+import org.openhab.binding.zwave.internal.protocol.ZWaveController;
+import org.openhab.binding.zwave.internal.protocol.ZWaveEndpoint;
+import org.openhab.binding.zwave.internal.protocol.ZWaveNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+
+/**
+ * Handles the Multi Command command class.
+ * 
+ * @author Hans Vanbellingen
+ * @since 1.7.0
+ */
+@XStreamAlias("crc16EncapsulationCommandClass")
+public class ZWaveCRC16EncapsulationCommandClass extends ZWaveCommandClass {
+
+	@XStreamOmitField
+	private static final Logger logger = LoggerFactory
+			.getLogger(ZWaveCRC16EncapsulationCommandClass.class);
+
+	/**
+	 * Creates a new instance of the ZWaveMultiCommandCommandClass class.
+	 * 
+	 * @param node
+	 *            the node this command class belongs to
+	 * @param controller
+	 *            the controller to use
+	 * @param endpoint
+	 *            the endpoint this Command class belongs to
+	 */
+	public ZWaveCRC16EncapsulationCommandClass(ZWaveNode node,
+			ZWaveController controller, ZWaveEndpoint endpoint) {
+		super(node, controller, endpoint);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public CommandClass getCommandClass() {
+		return CommandClass.CRC_16_ENCAP;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void handleApplicationCommandRequest(SerialMessage serialMessage,
+			int offset, int endpointId) {
+		logger.debug("NODE {}: Received CRC 16 Encapsulation Request", this
+				.getNode().getNodeId());
+		int command = serialMessage.getMessagePayloadByte(offset);
+		switch (command) {
+		case 0x01: // crc 16 Encapsulation
+			handleCRC16EncapResponse(serialMessage, offset + 1);
+			break;
+		}
+	}
+
+	/**
+	 * Handle the crc16 encapsulated message. This processes the received frame,
+	 * checks the crc and forwards to the real command class.
+	 * 
+	 * @param serialMessage
+	 *            The received message
+	 * @param offset
+	 *            The starting offset into the payload
+	 */
+	private void handleCRC16EncapResponse(SerialMessage serialMessage,
+			int offset) {
+		logger.trace("Process CRC16 Encapsulation");
+
+		// Check CRC
+		byte[] payload = serialMessage.getMessagePayload();
+		byte[] messageCrc = Arrays.copyOfRange(payload, payload.length - 2,
+				payload.length);
+		byte[] tocheck = Arrays.copyOfRange(payload, offset - 2,
+				payload.length - 2);
+
+		short calculatedCrc = crc_ccit(tocheck);
+		// check if messageCrc = calculatedCrc
+		ByteBuffer byteBuffer = ByteBuffer.allocate(2);
+		byteBuffer.putShort(calculatedCrc);
+		if (!Arrays.equals(messageCrc, byteBuffer.array())) {
+			logger.error("CRC not equal");
```

Comment:
I think it would be worth making this error a bit more useful. At least adding the node number so that if errors pop up in the log then someone can see if a particular device is generating lots of errors.